app/controllers/comments_controller.rb

```diff
@@ -19,6 +19,16 @@ def create
     end
   end
 
+  def destroy
+    @post = Post.includes(:comments).find(params[:post_id])
+    @comment = @post.comments.find(params[:id])
+    @comment.destroy
+    @post.comments_counter -= 1
+    @post.save
+    redirect_to("/users/#{current_user.id}/posts/#{@post.id}")
+    flash[:success] = 'Comment was deleted!'
+  end
+
```

Comment:
- [OPTIONAL] This method as implemented works. But, we could do things in more of `rails way`. Firstly, we don't need to find a Post with all the related comments, this is too much database operation for just a simple delete operation.

  Only find the comment we are interested in deleting. Since you also want to update the post, you can find the post through `@comment.post` and thereafter update the post as necessary. Please consider the suggestion below:
  
  ```ruby
  def destroy
    @comment = Comment.find(params[:id])
    @post = @comment.post
    @comment.destroy    
    @post.comments_counter -= 1
    @post.save
    redirect_to post_path(@post)
    flash[:success] = 'Comment was deleted!'
  end
  ```